src/init.luau

```diff
@@ -165,13 +189,20 @@ local function decode(refId: number, encoded: string, state: DecodeState)
 	elseif valueType == "Vector3" then
 		return Vector3.new(string.unpack("<nnn", value))
 	elseif valueType == "CFrame" then
-		return CFrame.new(string.unpack(`<{string.rep("nnn", 4)}`, value))
+		local cframeCase, pointer = string.unpack("<B", value)
+		if cframeCase ~= 0 then
+			local x, y, z = string.unpack("<nnn", value, pointer)
+			return CFrame.new(x, y, z) * CFRAME_SPECIAL_CASES[cframeCase]
```

Comment:
Instead of CFrame multiplication, let's use this instead:
```lua
local position = Vector3.new(string.unpack("<nnn", value, pointer))
local orientation = CFRAME_SPECIAL_CASES[cframeCase]
return CFrame.fromMatrix(position, orientation.XVector, orientation.YVector, orientation.ZVector)
```